### ISSUE TITLE
DiscordAPIError: Missing Permissions

### ISSUE THREAD

**Issue by naterchrdsn** [Spudnik-Group/Spudnik#156]
View details in Rollbar: [https://rollbar.com/Spudnik-Group/spudnik-bot-staging/items/2/](https://rollbar.com/Spudnik-Group/spudnik-bot-staging/items/2/)

```

DiscordAPIError: Missing Permissions
  File "/app/node_modules/discord.js/src/structures/Message.js", line 444, in CommandoMessage.delete
            .delete({ reason })
  File "/app/dist/modules/random/gitgud.js", line 39, in GitGudCommand.run
                msg.delete();
  File "/app/node_modules/discord.js-commando/src/extensions/message.js", line 214, in CommandoMessage.run
    				const promise = this.command.run(this, args, fromPattern);
  File "internal/process/next_tick.js", line 68, in process._tickCallback
```

**Comment by develogix** [comment]
@naterchrdsn should we consider writing a custom error handler that exposes `DiscordAPIError` exceptions to the user/guild where the command was called from? All other exception types should be thrown to Rollbar to be logged as a GitHub issue.

Linking #155, as it is related.

**Comment by naterchrdsn** [comment]
Yes, as this isn't a problem with the bot... it's a problem with the perms given to the bot.
We should have the default invite link add `Administrator` perms, at least.
But yes, for issues like this we need the bot to simply say: 'I don't have permission X'

**Comment by naterchrdsn** [comment]
Resolved in v0.5.0
